reader_concurrency_semaphore.cc

```diff
@@ -749,6 +749,23 @@ void reader_concurrency_semaphore::clear_inactive_reads() {
     }
 }
 
+future<> reader_concurrency_semaphore::evict_inactive_reads_for_table(table_id id) noexcept {
+    std::list<flat_mutation_reader_v2> evicted_readers;
+    auto it = _inactive_reads.begin();
+    while (it != _inactive_reads.end()) {
+        auto& ir = *it;
+        ++it;
+        if (ir.reader.schema()->id() == id) {
+            // No preemption-point in the filter loop, to avoid iteration invalidation.
+            evicted_readers.emplace_back(detach_inactive_reader(ir, evict_reason::manual));
```

Comment:
This violates noexcept.

But the handle is part of an intrusive list, detach unlinks it, so you could link it back to evicted_readers.

Reply:
Right. I will link them into another intrusive list, detaching them from handles.